### PR TITLE
Fixes #10824 - Refactoring errata index to show available errata to content view filter usin…

### PR DESCRIPTION
**Pull request by johnpmitsch**
…g a url parameter.

This is part of the conversion to scoped search for the filter_rules. By using a url parameter, the available errata is shown using the index of errata. This allows scoped search queries and autocomplete while searching the errata available to a content view filter

### CODE REVIEW

File: app/controllers/katello/concerns/api/v2/repository_db_content_controller.rb [hunk 1]
```diff
@@ -35,15 +35,20 @@ def compare
...
-        collection = filter_by_content_view_filter(@filter, collection) if @filter
         collection = filter_by_content_view_version(@version, collection) if @version
         collection = filter_by_environment(@environment, collection) if @environment
         collection = filter_by_repos(Repository.readable.in_organization(@organization), collection) if @organization
         collection = filter_by_ids(params[:ids], collection) if params[:ids]
+        if params[:available_for] == "content_view_filter"
+          collection = errata_available_for_content_view_filter(@filter, collection) if @filter
+        else
+          collection = filter_by_content_view_filter(@filter, collection) if @filter
+        end
```
Comment (jlsherrill) [inline app/controllers/katello/concerns/api/v2/repository_db_content_controller.rb]: This approach seems fine, but I'm not sure it belongs in the repository_db_content_controller.  Could we do something like:
```
if params[:available_for] == "content_view_filter" && self.responds_to?(:available_for_content_view_filter)
  colleection = self.available_for_content_view_filter(@filter, collection)
else
  filter_by_content_view_filter(@filter, collection) if @filter
end
```

and then define available_for_content_view_filter within the errata_controller?

Comment (johnpmitsch) [inline app/controllers/katello/concerns/api/v2/repository_db_content_controller.rb]: I can see how that would be much cleaner, thanks @jlsherrill just updated it

**Comment by johnpmitsch** [comment]
Refactoring to make changes to allow for autocomplete, please do not merge

File: app/controllers/katello/concerns/api/v2/repository_content_controller.rb [hunk 2]
```diff
@@ -150,7 +150,7 @@ def check_repo_for_content_resource
 
       def filter_by_content_view_filter(filter, options)
         ids = filter.send("#{singular_resource_name}_rules").map(&:uuid)
-        repo_ids = filter.applicable_repos.readable.select([:pulp_id, "#{Katello::Repository.table_name}.name"])
+        repo_ids = filter.applicable_repos.readable.map(&:pulp_id)
```
Comment (jlsherrill) [inline app/controllers/katello/concerns/api/v2/repository_content_controller.rb]: you might use  .pluck("#{Repository.table_name}.pulp_id")  as that should be faster

File: test/controllers/api/v2/errata_controller_test.rb [hunk 3]
```diff
@@ -10,6 +10,7 @@ def self.before_suite
     def models
       ::Katello::Product.any_instance.stubs(:as_json).returns([])
       @repo = Repository.find(katello_repositories(:rhel_6_x86_64))
+      @errata_filter = ContentViewFilter.find(katello_content_view_filters(:populated_erratum_filter))
```
Comment (jlsherrill) [inline test/controllers/api/v2/errata_controller_test.rb]: you can actually get rid of that ContentViewFilter.find() and just use:
```@errata_filter = katello_content_view_filters(:populated_erratum_filter))```

We used to have to do that, but not anymore

**Comment by jlsherrill** [comment]
Two very minor things, but otherwise looks good!

**Comment by johnpmitsch** [comment]
Thanks! you saw my note about not merging it? Actually refactoring it now

**Comment by johnpmitsch** [comment]
I had to change the filter controller to call Errata api to allow for autocomplete.

**Comment by jlsherrill** [comment]
APJ  :+1:

**Comment by johnpmitsch** [comment]
[test]

**Comment by jlsherrill** [comment]
ACK thanks @johnpmitsch !

**Pull request merged**